### PR TITLE
Translate higher order components

### PR DESCRIPTION
**Pull request by andriawan**
Dear maintainers,

Please review this translation. I have several concerns about some term which i am confused to translate. They are: 

- Subscribe
- data source
- Wrapped component
- inheritance
- container
- Children

Please help me solve that @resir014 @regalius

### CODE REVIEW

**Comment by resir014** [comment]
@andriawan 

- Children: let's keep it as *children*.
- Subscription can be translated to "langganan" or any variations of it.
- Data source: can be translated to "sumber data", unless part of a code (e.g. `SumberData`)
- Wrapped component: feel free to translate where possible, otherwise fallback to *wrapped component*
- Inheritance: keep it (*inheritance*)
- Container: keep it (*container*)

**Comment by andriawan** [comment]
thanks for the feedback @resir014...i will re-review my work based on your suggestion above

File: content/docs/higher-order-components.md [hunk 1]
```diff
@@ -314,60 +314,60 @@ function getDisplayName(WrappedComponent) {
...
+Komponen HOC datang dengan beberapa batasan yang kurang jelas jika Anda baru menggunakan React.
 
-### Don't Use HOCs Inside the render Method {#dont-use-hocs-inside-the-render-method}
+### Jangan Menggunakan HOC di dalam Metode _render_ {#dont-use-hocs-inside-the-render-method}
 
-React's diffing algorithm (called reconciliation) uses component identity to determine whether it should update the existing subtree or throw it away and mount a new one. If the component returned from `render` is identical (`===`) to the component from the previous render, React recursively updates the subtree by diffing it with the new one. If they're not equal, the previous subtree is unmounted completely.
+Algoritma _React's diffing_ (disebut _reconciliation_) menggunakan identitas komponen untuk menentukan apakah subtree yang ada perlu diperbarui atau _mount_ yang baru. Jika komponen yang dikembalikan dari `render` sama (`===`) dengan _render_ komponen sebelumnya, React memperbarui subtree secara rekursif dengan membandingkan dengan yang baru. Jika tidak sama, subtree sebelumnya akan diganti seluruhnya.
```
Comment (resir014) [inline content/docs/higher-order-components.md]:
```suggestion
Algoritma _diffing_ React (disebut _reconciliation_) menggunakan identitas komponen untuk menentukan apakah subtree yang ada perlu diperbarui atau _mount_ yang baru. Jika komponen yang dikembalikan dari `render` sama (`===`) dengan _render_ komponen sebelumnya, React memperbarui subtree secara rekursif dengan membandingkan dengan yang baru. Jika tidak sama, subtree sebelumnya akan diganti seluruhnya.
```

File: content/docs/higher-order-components.md [hunk 2]
```diff
@@ -4,53 +4,53 @@ title: Higher-Order Components
...
 
-Concretely, **a higher-order component is a function that takes a component and returns a new component.**
+Konkritnya, **_higher-order component_ merupakan fungsi yang mengambil sebuah komponen dan mengembalikan sebuah komponen baru**
 
 ```js
 const EnhancedComponent = higherOrderComponent(WrappedComponent);
 ```
 
-Whereas a component transforms props into UI, a higher-order component transforms a component into another component.
+Sebaliknya saat sebuah komponen mengubah _props_ menjadi antar muka pengguna (UI), _higher-order component_ mengubah sebuah komponen menjadi komponen yang lainnya.
```
Comment (resir014) [inline content/docs/higher-order-components.md]:
```suggestion
Sebaliknya saat sebuah komponen mengubah _props_ menjadi antarmuka pengguna (UI), _higher-order component_ mengubah sebuah komponen menjadi komponen yang lainnya.
```

"antarmuka" - digabung.

File: content/docs/higher-order-components.md (hunk 1 above)
Comment (andriawan) [reply content/docs/higher-order-components.md]: got it...i'll fix it

File: content/docs/higher-order-components.md (hunk 2 above)
Comment (andriawan) [reply content/docs/higher-order-components.md]: okay, thanks for the feedback

**Comment by andriawan** [comment]
please check again @resir014, i have made several change based on your suggestion

**Pull request merged**